File: authentication-service/auth_service.py
from flask import Flask, request, jsonify

app = Flask(__name__)

# This is a mock user database. In a real-world scenario, you'd use a proper database.
users = {
    "user1": "password1",
    "user2": "password2"
}

@app.route('/validate', methods=['POST'])
def validate():
    data = request.json
    userid = data.get('userid')
    password = data.get('password')
    
    if userid in users and users[userid] == password:
        return jsonify({"message": "Authentication successful", "userid": userid}), 200
    else:
        return jsonify({"error": "Invalid credentials"}), 401

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=8000)

File: authentication-service/auth_service.js
const express = require("express");
const bodyParser = require("body-parser");

const app = express();
const port = process.env.AUTH_PORT || 8000;

app.use(bodyParser.json());

// This is a mock user database. In a real-world scenario, you'd use a proper database.
const users = {
  user1: "password1",
  user2: "password2",
};

app.post("/validate", (req, res) => {
  const { userid, password } = req.body;

  if (users[userid] && users[userid] === password) {
    res
      .status(200)
      .json({ message: "Authentication successful", userid: userid });
  } else {
    res.status(401).json({ error: "Invalid credentials" });
  }
});

app.listen(port, () => {
  console.log(`Authentication service listening at http://localhost:${port}`);
});
